llpc/patch/llpcPatchCheckShaderCache.cpp

```diff
@@ -96,6 +96,35 @@ bool PatchCheckShaderCache::runOnModule(
 
     Patch::Init(&module);
 
+    // NOTE: Global constant are added to the end of pipeline binary. we can't merge ELF binaries if global constant
+    // is used in non-fragment shader stages.
+    for (auto& global : module.globals())
+    {
+        if (auto pGlobalVar = dyn_cast<GlobalVariable>(&global))
+        {
+            if (pGlobalVar->isConstant())
+            {
+                SmallVector<const Value*, 4> vals;
+                vals.push_back(pGlobalVar);
+                for (uint32_t i = 0; i != vals.size(); ++i)
+                {
+                    for (auto pUser : vals[i]->users())
+                    {
+                        if (isa<Constant>(pUser))
+                        {
+                            vals.push_back(pUser);
```

Comment:
This code seems to be only writing to the vector. Do we need `vals` at all?

Reply:
The lines above read it. It's a worklist algorithm.